### PR TITLE
config(amazonq): inline project context ab

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Problem
update inline suggestion project context ab test config

#### CONTROL
- use bm25 opentabs context 

#### TREATMENT_1
- use repomap + bm25 open tabs context 

#### TREATMENT_2
- use bm25 global project context 

## Solution

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ interface Chunk {
     score?: number
 }
 
-type SupplementalContextConfig = 'none' | 'v1' | 'v2'
+type SupplementalContextConfig = 'none' | 'openbtabs' | 'codemap' | 'bm25' | 'default'
```
Comment (leigaol) [inline packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts:56]: typo

**Review by leigaol — APPROVED** [review]

File: packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts [hunk 2]
```diff
@@ -64,43 +64,114 @@ export async function fetchSupplementalContextForSrc(
...
+    // fallback to opentabs if projectContext timeout for 'default' | 'bm25'
+    const opentabsContextPromise = waitUntil(
         async function () {
-            return await fetchSupplementalContextForSrcV1(editor, cancellationToken)
+            return await fetchOpentabsContext(editor, cancellationToken)
         },
         { timeout: supplementalContextTimeoutInMs, interval: 5, truthy: false }
     )
-    const promiseV2 = waitUntil(
+
```
Comment (leigaol) [inline packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts:108]: this is for global bm25 context only, it fallback to open tab bm25

File: packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts [hunk 3]
```diff
@@ -64,43 +64,114 @@ export async function fetchSupplementalContextForSrc(
...
-        return fetchSupplementalContextForSrcV1(editor, cancellationToken)
+
+    if (supplementalContextConfig === 'openbtabs') {
+        return {
+            supplementalContextItems: (await fetchOpentabsContext(editor, cancellationToken)) ?? [],
+            strategy: 'OpenTabs_BM25',
+        }
+    }
+
+    if (supplementalContextConfig === 'codemap') {
```
Comment (leigaol) [inline packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts:75]: this is codemap + open tab bm25

**Comment by justinmk3** [comment]
This correlates with test failures now seen on master (note: there are 2 different test runs; the failure is in the first, near the top of the build logs; search for the string "passing" to find it):

- https://github.com/aws/aws-toolkit-vscode/actions/runs/11842327301/job/33000682150
- https://github.com/aws/aws-toolkit-vscode/actions/runs/11842216947/job/33000324695
- https://github.com/aws/aws-toolkit-vscode/actions/runs/11842216947/job/33000325796
- https://github.com/aws/aws-toolkit-vscode/actions/runs/11842216947/job/33000323199

```
  1) crossFileContextUtil
       fetchSupplementalContextForSrc
         for control group, should return opentabs context where there will be 3 chunks and each chunk should contains 50 lines:

      AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

  assert_1.default.ok(actual.supplementalContextItems.length === 3)

      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (test/unit/codewhisperer/util/crossFileContextUtil.test.ts:51:20)
```

**Pull request merged**